### PR TITLE
Correct vpg

### PR DESCRIPTION
**Pull request by Sharad24**
Resolves some issues created yesterday during refactoring of VPG e.g. missing `get_traj_loss` definition. And resolves a part of #60

### CODE REVIEW

**Comment by Sharad24** [comment]
Not sure if there is a mistake here. Does not seem to train.

File: jigglypuffRL/common/base.py [hunk 1]
```diff
@@ -15,22 +15,22 @@ def __init__(self, discrete, deterministic, **kwargs):
         self.model = None
 
     def forward(self, state):
-        return self.model(state)
+        return self.model.forward(state)
```
Comment (Het-Shah) [inline jigglypuffRL/common/base.py:18]: Why is this necessary?

**Comment by Het-Shah** [comment]
> missing get_traj_loss definition. 
How did this go missing?

**Comment by Sharad24** [comment]
Got removed in #66 
Not sure why + I missed it too while reviewing.

File: jigglypuffRL/deeprl/vpg/vpg.py [hunk 2]
```diff
@@ -149,26 +148,64 @@ def create_model(self):
 
 
     def select_action(self, state):
-        state = torch.as_tensor(state).float().to(self.device)
+        state = Variable(torch.as_tensor(state).float().to(self.device))
 
         # create distribution based on policy_fn output
         a, c = self.ac.get_action(state)
-        val = self.ac.get_value(state)
+        val = self.ac.get_value(state).unsqueeze_(0)
```
Comment (Het-Shah) [inline jigglypuffRL/deeprl/vpg/vpg.py:155]: this should be `unsqueeze(0)`

File: jigglypuffRL/common/base.py (hunk 1 above)
Comment (Sharad24) [reply jigglypuffRL/common/base.py:18]: Not necessary at all, was trying to debug through some errors. So just though explicitly stating it would be better. 

We can reset this change though

File: jigglypuffRL/deeprl/vpg/vpg.py (hunk 2 above)
Comment (Sharad24) [reply jigglypuffRL/deeprl/vpg/vpg.py:155]: Right, thanks

File: jigglypuffRL/common/base.py (hunk 1 above)
Comment (Het-Shah) [reply jigglypuffRL/common/base.py:18]: This might cause errors, that is why I pointed this out.

Comment (Sharad24) [reply jigglypuffRL/common/base.py:18]: Why?

Comment (Het-Shah) [reply jigglypuffRL/common/base.py:18]: https://discuss.pytorch.org/t/any-different-between-model-input-and-model-forward-input/3690/4

File: jigglypuffRL/deeprl/vpg/vpg.py [hunk 3]
```diff
@@ -149,26 +148,64 @@ def create_model(self):
...
         self.policy_hist = torch.cat(
             [self.policy_hist, c.log_prob(a).unsqueeze(0)]
         )
 
+        self.value_hist = torch.cat([self.value_hist, val])
+
         # clear traj history
         self.traj_reward = []
         self.policy_hist = Variable(torch.Tensor())
         self.value_hist = Variable(torch.Tensor())
```
Comment (ajaysub110) [inline jigglypuffRL/deeprl/vpg/vpg.py:167]: This should not be done within `select_action`, only in `get_traj_loss()`. No clue how these errors are coming up now

File: jigglypuffRL/common/base.py (hunk 1 above)
Comment (Sharad24) [reply jigglypuffRL/common/base.py:18]: I know forward just uses `__call__` but why would this create errors?

File: jigglypuffRL/deeprl/vpg/vpg.py (hunk 3 above)
Comment (Sharad24) [reply jigglypuffRL/deeprl/vpg/vpg.py:167]: Lol, well this is why its not training

**Comment by ajaysub110** [comment]
Is it training now?

**Comment by Sharad24** [comment]
Have not tried yet….

**Comment by Sharad24** [comment]
Nope does not work. Its starting to look like a silly error somewhere
